src/coreclr/jit/lower.cpp

```diff
@@ -6900,7 +6900,8 @@ bool Lowering::NodesAreEquivalentLeaves(GenTree* tree1, GenTree* tree2)
 bool Lowering::CheckMultiRegLclVar(GenTreeLclVar* lclNode, const ReturnTypeDesc* retTypeDesc)
```

Comment:
I fixed the logic of `CheckMultiRegLclVar()` the way you suggested. I am not fully sure about this example though. Can you give C# example where I can see things goes wrong?

Reply:
> Can you give C# example where I can see things goes wrong?

It is not unlikely this cannot actually be reproduced right now because of the constrained way multi-reg locals are used, but here would be the idea for a reproduction:
```
// V00: struct { long, long }
// V01: struct { int, int, int, int }

V00 = DivDem()
*(typeof(V00)*)V01 = V00
```
Then V00 and V01 need to be promoted and forward subtitution needs to happen s.t. we end with:
```
V01 = DivRem()
```
The more general point though is that this _can_ happen according the IR validity rules and so needs to be handled.